sources/packages/api/src/msfaa-integration/msfaa-request.service.ts

```diff
@@ -0,0 +1,151 @@
+  MSFAARecord,
+  MSFAAUploadResult,
+} from "./models/msfaa-integration.model";
+import { MSFAAIntegrationService } from "./msfaa-integration.service";
+
+@Injectable()
+export class MSFAARequestService {
+  constructor(
+    private readonly msfaaNumberService: MSFAANumberService,
+    private readonly msfaaService: MSFAAIntegrationService,
+    private readonly sequenceService: SequenceControlService,
+  ) {}
+
+  /**
+   * 1. Fetches the MSFAA records which are not sent for request.
+   * 2. Create Unique sequence for the request sent file.
+   * 3. Create the Request content for the MSFAA file by populating the
+   *      header, footer and trailer content.
+   * 4. Create the request filename with the file path for the MSFAA Request
+   *      sent File.
+   * 5. Upload the content to the zoneB SFTP server.
+   * 6. Update the MSFAA records, that are sent in the request sent file.
+   * @returns Processing MSFAA request result.
+   */
+  async processMSFAARequest(
+    offeringIntensity: string,
+  ): Promise<MSFAAUploadResult> {
+    this.logger.log(`Retrieving pending ${offeringIntensity} MSFAA request...`);
+    const pendingMSFAARequests =
+      await this.msfaaNumberService.getPendingMSFAARequest(offeringIntensity);
+    if (!pendingMSFAARequests.length) {
+      return {
+        generatedFile: "none",
+        uploadedRecords: 0,
+      };
+    }
+    this.logger.log(
+      `Found ${pendingMSFAARequests.length} MSFAA number(s) for ${offeringIntensity} application that needs request.`,
+    );
+    const msfaaRecords = pendingMSFAARequests.map((pendingMSFAARequest) => {
+      return this.createMSFAARecord(pendingMSFAARequest, offeringIntensity);
+    });
+
+    //Fetches the MSFAANumber ids, for further update in the db.
+    const msfaaRecordIds = pendingMSFAARequests.map(
+      (pendingMSFAARequest) => pendingMSFAARequest.id,
+    );
+
+    //Total hash of the Student's SIN, its used in the footer content.
+    const totalSINHash = pendingMSFAARequests.reduce(
+      (accumulator, pendingMSFAARequest) =>
+        accumulator + parseInt(pendingMSFAARequest.student.sin),
+      0,
+    );
+
+    //Create records and create the unique file sequence number
+    let uploadResult: MSFAAUploadResult;
+    await this.sequenceService.consumeNextSequence(
+      `MSFAA_${offeringIntensity}_SENT_FILE`,
+      async (nextSequenceNumber: number, entityManager: EntityManager) => {
+        try {
+          this.logger.log("Creating MSFAA request content...");
+          // Create the Request content for the MSFAA file by populating the
+          // header, footer and trailer content.
+          const fileContent = this.msfaaService.createMSFAARequestContent(
+            msfaaRecords,
+            nextSequenceNumber,
+            totalSINHash,
+          );
+          // Create the request filename with the file path for the MSFAA Request
+          // sent File.
+          const fileInfo = await this.msfaaService.createRequestFileName(
```

Comment:
As we talked about, this method is creating a new transaction that will not fail together with the rest of this block.
As we discussed, lets try to pass the entityManager to the nested consumeNextSequence to make everything part of the same transation.

Reply:
Good work refactoring it and extending the functionality 😉